### PR TITLE
Fix object deletion where quorum is not available

### PR DESCRIPTION
**Pull request by nitisht**
## Description
Object deletion should not be possible if quorum is not
available. This PR updates deleteObject() to check for
quorum errors before proceeding with object deletion.

## Motivation and Context
See #5535

## How Has This Been Tested?
Manually with `mc rm`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

File: cmd/xl-v1-object.go [hunk 1]
```diff
@@ -730,49 +730,54 @@ func (xl xlObjects) deleteObject(bucket, object string) error {
...
 	if !isDir {
 		// Read metadata associated with the object from all disks.
 		metaArr, errs := readAllXLMetadata(xl.getDisks(), bucket, object)
-
 		// get Quorum for this object
 		_, writeQuorum, err = objectQuorumFromMeta(xl, metaArr, errs)
 		if err != nil {
 			return err
 		}
+		err = reduceWriteQuorumErrs(errs, objectOpIgnoredErrs, writeQuorum)
```
Comment (vadmeste) [inline cmd/xl-v1-object.go:741]: This is little hard to read, the reason is that listObjectModtimes() requires the parameter 'errs' but we don't really need it.. we can consider 'nil' in metaArray as an error. I can fix it in another PR.

File: cmd/xl-v1-object.go [hunk 2]
```diff
@@ -730,49 +730,54 @@ func (xl xlObjects) deleteObject(bucket, object string) error {
...
 	} else {
+		// WriteQuorum is defaulted to N/2 + 1 for directories
 		writeQuorum = len(xl.getDisks())/2 + 1
 	}
 
-	// Initialize list of errors.
-	var dErrs = make([]error, len(xl.getDisks()))
+	// No error, means proceed with deletion request
+	if err == nil {
```
Comment (vadmeste) [inline cmd/xl-v1-object.go]: Can we return here if err is set so we can avoid a indent level in the next block ?

Comment (nitisht) [reply cmd/xl-v1-object.go]: good idea, done @vadmeste

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Review by harshavardhana — APPROVED** [review]
LGTM

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**